src/realm/column_timestamp.hpp

```diff
@@ -126,20 +126,21 @@ class TimestampColumn : public ColumnBaseSimple {
     template <class Condition>
     Timestamp minmax(size_t* result_index) const noexcept
     {
-        // Condition is realm::Greater for maximum and realm::Less for minimum.
-
+        // Condition is realm::Greater for maximum and realm::Less for minimum. Any non-null value is both larger
+        // and smaller than a null value.
         if (size() == 0) {
             if (result_index)
                 *result_index = npos;
             return Timestamp{};
         }
 
         Timestamp best = get(0);
-        size_t best_index = 0;
+        size_t best_index = !best.is_null() ? 0 : npos; // npos if null
```

Comment:
Don't not not avoid changing the negation for readability, as suggested by Daniel :-)